### ISSUE TITLE
PMT transactions need to create their own transaction number (EMP_CD_CSHR and DOC_SEQ_NUM)

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#488]
_Imported from trac ticket 488.
Created by **kbower**
Opened in trac: Dec 11, 2010, 8:58:16 AM
Last modified in trac: Dec 27, 2010, 10:48:16 AM_
***
_by kbower:_
**We need to set `EMP_CD_CSHR` and `DOC_SEQ_NUM` sometimes**

**From Tim**:
_PMT transactions need to create their own transaction number. Re: Order # 1210004AC24.  PMT on this order shows 1210004 with nothing appended: we need to be able to access PMT type payments for viewing and correction, and need a transaction number to do so._

**From Kent:**
_Wayne,
     Under what circumstances is the AR_TRN field 'DOC_SEQ_NUM' set?   
It appears to be blank for "SAL", but not for some other TRN_TP_CDs.
_
Thank you,
Kent_
_

**From Wayne**:
_Kent,
_
The AR_TRN.DOC_SEQ_NUM is the four alphanumeric numbers after the store
code in the document number however it is only consistently applied
where AR_TRN.TRN_TP_CD in ('R', 'DEP', 'PMT') and inconsistently where
AR_TRN.TRN_TP_CD in ('MCR', 'MDB', 'SER'). We can set AR_TRN.DOC_SEQ_NUM
in all cases.

Thanks,
 
Wayne Liburd
_
_

**Comment by kbower** [comment]
_**kbower** trac comment on Dec 11, 2010, 9:22:19 AM:_

When I issue this sql:
```
SQL> select * from ar_trn where TRN_TP_CD = 'PMT' and create_dt > sysdate - 20;
```
I get these results.  It looks like 2 might be from us and 2 from GERS.  I think the 
```
CO_ CUST_CD    CNTR_CD    MOP EMP_CD_CSH EMP_CD_APP EMP_CD_OP  OR CSH TRN BNK BAT_DOC_NU B DOC_        AMT
--- ---------- ---------- --- ---------- ---------- ---------- -- --- --- --- ---------- - ---- ----------
POST_DT              S A BNK_NUM    CHK_NUM    IVC_CD         BNK_CRD_NUM          APP_CD
-------------------- - - ---------- ---------- -------------- -------------------- ----------
DES                                      ADJ_IVC_CD       CHNG_OUT PM ARBR_DT              WR_DT
---------------------------------------- -------------- ---------- -- -------------------- --------------------
POST_ID_NUM ORIGI EXP_DT               REF_NUM  HOST_REF B CREATE_DT             AR_TRN_PK     LN_NUM
----------- ----- -------------------- -------- -------- - -------------------- ---------- ----------
ACCT_NUM
----------------------------------------
KFC 27                    AMX                                  04 04  PMT                              107
10-DEC-2010 00:00:00 T O                       1210004AC29
                                                                 0
                                                           10-DEC-2010 16:51:46    7793589


KFC RIVED54236            CK                                   04 04  PMT                             53.5
10-DEC-2010 00:00:00 T O                       1210004AC24
                                                                 0
                                                           10-DEC-2010 11:46:32    7793541


KFC LOPED54327            MC  SLSTRN                SLSTRN     09 09  PMT                  AMEI    1376.15
23-NOV-2010 00:00:00 T O                       1123009AMEC
                                                                 0 09
            POA                                            23-NOV-2010 10:20:02    7792887


KFC LOPED54327            DC  SLSTRN                SQLTRAIN   09 09  PMT                  AMEL     889.64
23-NOV-2010 00:00:00 T O                       1123009AMECA
                                                                 0 09
            POA                                            23-NOV-2010 11:06:05    7792891

```
To me, it looks like we aren't setting `EMP_CD_CSHR` or `DOC_SEQ_NUM`.

**Comment by kbower** [comment]
_**kbower** trac comment on Dec 11, 2010, 9:26:08 AM:_

Actually, #355 set `DOC_SEQ_NUM = SO.SO_SEQ_NUM`, which is apparently (sometimes at least) incorrect.

It looks like gers is setting this field more often than we:

```
IVC_CD         TRN MOP DOC_
-------------- --- --- ----

1208014AMPA    DEP MC  AMPA
1206015AMJW    DEP AMX AMJW
1206015AMJW    SAL
1206015AMKB    SAL FI
1206015AMKI    SAL FI
1206015AMKJ    SAL FI
1207000AC18    DEP CS
1207000AC18    SAL
1210004AC19    DEP CS
1210004AC19    SAL
1210004AC24    DEP AMX
1210004AC24    SAL
1210004AC24    PMT CK
1210004AC26    DEP MC
1210004AC26    SAL
1210000AC27    DEP CS
1210000AC27    SAL
1210004AC29    DEP CS
1210004AC29    DEP COD
1210004AC29    SAL
1206018AMJB    DEP AMX AMJB
1206018AMJB    SAL
1206018AMJF    DEP DC  AMJF
1206018AMJF    SAL
1206018AMJL    DEP MC  AMJL
1206018AMJL    SAL
1206018AMJU    DEP DC  AMJU
1206018AMJU    SAL
1206018AMKE    DEP CK  AMKE
1206018AMKE    DEP AMX AMKE
1206018AMKE    SAL
```

**Comment by kbower** [comment]
_**kbower** trac comment on Dec 24, 2010, 7:16:18 AM:_

(In [[989]](../commit/604502f3599e5ed209ebb53010cd8f5654f467c1))
See #488.

**Comment by kbower** [comment]
_**rneville** trac comment on Dec 27, 2010, 10:48:16 AM:_

(In [[995]](../commit/926c983f2c3620f434fd3c1b96c4debe97d6c688))
closes #488 and cleans up some junk with _newpayments

**Comment by kbower** [comment]
_**rneville** trac status change on Dec 27, 2010, 10:48:16 AM:_ **closed**
